libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxManeuverView.kt

```diff
@@ -230,10 +230,10 @@ class MapboxManeuverView : ConstraintLayout {
         replaceWith = ReplaceWith("renderManeuverWith(shields)")
     )
     fun renderManeuverShields(shieldMap: Map<String, RoadShield?>) {
-        shieldMap.forEach { entry ->
-            ifNonNull(entry.value) { value ->
-                routeShields.add(value.toRouteShield())
-            }
+        val shields = shieldMap.values.filterNotNull().map(RoadShield::toRouteShield)
+        synchronized(routeShields) {
+            routeShields.clear()
+            routeShields.addAll(shields)
```

Comment:
`routeShields` is also accessed in `renderManeuvers`, why is there no `synchronized`? I see the same issue in `MapboxRoadNameView`.  
But anyway it is weird to have synchronization in a view, because updates are not allowed from background threads.

Reply:
If anyone tries to call `renderManeuverShields` from a background thread, they will get a crash, so there is no point in having synchronization here.

Reply:
I tend to agree that synchronized isn't necessary because all the calls should be made on the main thread given it's a view class. However, it was added in response to feedback expressed above so I added it. Since I'm not the original author of the class I didn't get strongly opinionated on this issue.

Reply:
@abhishek1508 what are your thoughts?